deployment/environment.go

```diff
@@ -76,6 +76,17 @@ type Environment struct {
 	Offchain          OffchainClient
 }
 
+func (e Environment) Copy() (Environment, error) {
```

Comment:
I think this is confusing because its only does a deep copy of one field. Its more clear to just use the NewEnvironment constructor in ApplyChangesets and remove this